src/plugins/intel_npu/tests/unit/npuw/set_tensor.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include "openvino/openvino.hpp"
+#include "npuw/compiled_model.hpp"
+#include "npuw/just_sync_infer_request.hpp"
+#include "model_generator/model_generator.hpp"
+#include "plugin.hpp"
+
+// FIXME: parametrize all the tests below
+
+// TODO: add tests on Unfold, Base and Just for inputs and outputs (where applicable)
+TEST(SetTensor, RemoteTensorOutputJust) {
+    // Only run this test on NPU device
+    ov::Core ov_core;
+    auto core_devices = ov_core.get_available_devices();
+    if (std::find(core_devices.begin(), core_devices.end(), "NPU") == core_devices.end()) {
+        GTEST_SKIP() << "No available devices.";
+    }
+
+    // Create model
+    ModelGenerator mg;
+    auto model = mg.get_model_with_repeated_blocks();
+
+    ov::element::Type element_type = ov::element::i32;
+    auto output_tensor_shape = model->outputs()[0].get_shape();
+    // Calculate total number of elements
+    size_t total_elements = ov::shape_size(output_tensor_shape);
+
+    // Create output data
+    std::vector<int> data = std::vector<int>(total_elements, 0);
+    std::iota(data.begin(), data.end(), 1);
+
+    // Create the remote tensor output
+    auto npu_context = ov_core.get_default_context("NPU");
+    auto output = npu_context.create_host_tensor(element_type, output_tensor_shape);
+
+    // Initialize remote input with non-zero data
+    ov::Tensor values(element_type, output_tensor_shape, data.data());
+    values.copy_to(output);
+
+    // Create plugin object
+    auto plugin = std::make_shared<intel_npu::Plugin>();
+
+    // Compile NPUW
+    auto compiled = std::make_shared<ov::npuw::CompiledModel>(model, plugin, ov::AnyMap{});
+
+    // Create infer request
+    std::shared_ptr<ov::ISyncInferRequest> request;
+    request = std::make_shared<ov::npuw::JustInferRequest>(compiled);
```

Comment:
Alternative idea how to test `set_tensor`:
1) Use default OV API (compile_model, create_infer_request, etc), but set NPUW properties (e.g. UNFOLD)
2) To test output tensor can just check get_tensor produces non-empty tensor after set
3) To test input tensor - TBD